### PR TITLE
feat(JOML): migrate joml for BlockRegion

### PR DESCRIPTION
**Pull request by jdrueckert**
PR: https://github.com/MovingBlocks/Terasology/pull/4296

### CODE REVIEW

File: src/main/java/org/terasology/structureTemplates/internal/systems/ProtectedRegionClientSystem.java [hunk 1]
```diff
@@ -58,7 +58,8 @@ private boolean isInProtectedRegion(Collection<Vector3i> positions) {
 
     @ReceiveEvent(priority = EventPriority.PRIORITY_CRITICAL, components = {NoInteractionWhenProtected.class})
     public void onActivationPredicted(ActivationPredicted event, EntityRef target) {
-        Vector3f position = event.getTarget().getComponent(LocationComponent.class).getWorldPosition();
+        Vector3f position = event.getTarget().getComponent(LocationComponent.class).getWorldPosition(new Vector3f());
+        //TODO: is this equivalent to  new Vector3i(position, RoundingMode.HALF_UP);
```
Comment (skaldarnar) [inline src/main/java/org/terasology/structureTemplates/internal/systems/ProtectedRegionClientSystem.java:62]: @pollend I think we should change this in the future to use the conversion with rounding mode. Leaving this as is for now to be on the safe side.

File: src/main/java/org/terasology/structureTemplates/internal/systems/ProtectedRegionServerSystem.java [hunk 2]
```diff
@@ -92,22 +94,23 @@ private boolean isInProtectedRegion(Collection<Vector3i> positions) {
...
     @ReceiveEvent(priority = EventPriority.PRIORITY_CRITICAL)
     public void onPlaceBlocks(PlaceBlocks event, EntityRef entity) {
         EntityRef instigator = event.getInstigator();
         EntityRef player = instigator.getOwner();
         if (!player.hasComponent(ClientComponent.class)) {
             return;
         }
-        if (isInProtectedRegion(event.getBlocks().keySet())) {
+
+        if (isInProtectedRegion(event.getBlocks().keySet().stream().map(JomlUtil::from).collect(Collectors.toSet()))) {
```
Comment (jdrueckert) [inline src/main/java/org/terasology/structureTemplates/internal/systems/ProtectedRegionServerSystem.java]: This is hard to read @skaldarnar can we break that up a bit with descriptive var names?

**Review by jdrueckert — COMMENTED** [review]
Yup that's better now. Would approve if I could (PR creator and stuff).

**Review by skaldarnar — APPROVED** [review]

File: src/main/java/org/terasology/structureTemplates/internal/systems/ProtectedRegionClientSystem.java [hunk 3]
```diff
@@ -58,7 +58,8 @@ private boolean isInProtectedRegion(Collection<Vector3i> positions) {
...
         Vector3i roundedPosition = new Vector3i(Math.round(position.x), Math.round(position.y), Math.round(position.z));
```
Comment (pollend) [inline src/main/java/org/terasology/structureTemplates/internal/systems/ProtectedRegionClientSystem.java:63]: there are some changes for rounding that would be relevant. 
https://github.com/JOML-CI/JOML/blob/main/src/org/joml/Vector3i.java#L137-L141

**Pull request merged**